Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/NoRegionsInMethodAnalyzer.cs

```diff
@@ -28,12 +26,11 @@ public class NoRegionsInMethodSyntaxNodeAction : SyntaxNodeAction<MethodDeclarat
 		public override void Analyze()
 		{
 			// Specifying Span instead of FullSpan correctly excludes trivia before or after the method
-			var descendants = Node.DescendantNodes(Node.Span, null, descendIntoTrivia: true);
+			var descendants = Node.DescendantNodes(Node.Span, null, descendIntoTrivia: true).OfType<DirectiveTriviaSyntax>();
```

Comment:
We're calling this twice?

Reply:
The performance gain here, is that the filtering is done earlier. That seemed to help speed up the foreach's that follow

Reply:
Ahhh I missed DirectiveTriviaSyntax vs RegionDirectiveTriviaSyntax.